src/select/select.tsx

```diff
@@ -145,64 +124,104 @@ class Select<T> extends React.Component<Props<T>, State<T>> {
+                onClick={this.toggleOptionsList}
+            >
+                <span className={`text`} data-qa={`select-toggle--standard-text-${this.props.id}`}>
+                    {label || this.props.placeholder}
+                </span>
+                <span className="select-option-toggle--icon-dropdown" />
+            </div>
+        );
     }
 
-    private isSelected(option: Option<T>) {
-        return isEqual(this.props.value, option.value);
+    private renderSearchableTrigger() {
+        const label = this.getLabel();
+        const disabledClass = this.props.disabled ? ' disabled' : '';
+        const placeholderClass = this.state.isExpanded || !label ? ' has-faded-placeholder' : '';
+        return (
+            <div
+                role="button"
+                data-qa={`select-toggle--search-${this.props.id}`}
+                className={`select-option-toggle select-option-toggle--search${disabledClass}${placeholderClass}`}
+                aria-haspopup="listbox"
+            >
+                {this.getSearchInput()}
+            </div>
+        );
     }
 
-    private filterOptions(event: React.ChangeEvent<HTMLInputElement>) {
-        const filteredOptions = this.props.options.filter((option) => matches(event.target.value, option.label));
-        this.setState({
-            options: filteredOptions,
-        });
+    private getSearchInput() {
+        return (
+            <>
+                <span className="select-option-toggle--icon-filter" />
+                <input
+                    id={this.props.id}
+                    type="text"
+                    className={`text`}
+                    autoComplete="off"
+                    onChange={this.onFilterChange}
+                    value={this.state.filterValue}
+                    disabled={this.props.disabled}
+                    onClick={this.expandOptionsList}
+                    placeholder={this.getSearchTogglePlaceholder()}
+                    data-qa={`select-option-toggle--input-${this.props.id}`}
+                />
+                <button
+                    type="button"
+                    className="select-option-toggle--icon-clear"
+                    onClick={this.clearFilterValue}
+                    disabled={!this.state.filterValue.length || this.props.disabled}
+                    data-qa={`select-option-toggle--clear-${this.props.id}`}
+                />
+                <button
+                    type="button"
+                    className="select-option-toggle--icon-dropdown"
+                    onClick={this.toggleOptionsList}
+                    data-qa={`select-option-toggle--dropdown-${this.props.id}`}
+                    disabled={this.props.disabled}
+                />
+            </>
+        );
     }
 }
 
-const StringSelect = Select as new () => Select<string>;
-const NumberSelect = Select as new () => Select<number>;
+const StringSelect = Select as new () => Select<string, Option<string>>;
```

Comment:
you will be able to avoid `, Option<string>` for these two after you added the default

Reply:
Those two could be removed. They are not used anywhere, and honestly, I don't see the point of having them.

Reply:
Convenience. ;) Anyway